semantic-core/src/Control/Carrier/Readline/Haskeline.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, ScopedTypeVariables, TypeApplications, TypeOperators, UndecidableInstances #-}
+module Control.Carrier.Readline.Haskeline
```

Comment:
This has been split out of `Control.Effect.Readline`, and significantly streamlined.